cli/cmd/install.go

```diff
@@ -205,56 +219,155 @@ func validateAndBuildConfig(options *installOptions) (*installConfig, error) {
 	}, nil
 }
 
+func (options *installOptions) validate() error {
+	if _, err := log.ParseLevel(options.controllerLogLevel); err != nil {
+		return fmt.Errorf("--controller-log-level must be one of: panic, fatal, error, warn, info, debug")
+	}
+
+	if options.proxyAutoInject && options.singleNamespace {
+		return fmt.Errorf("The --proxy-auto-inject and --single-namespace flags cannot both be specified together")
+	}
+
+	return options.proxyConfigOptions.validate()
+}
+
 func render(config installConfig, w io.Writer, options *installOptions) error {
-	template, err := template.New("linkerd").Parse(install.Template)
+	// Load chart for rendering
+	chrt, err := loadChart(&config)
 	if err != nil {
 		return err
 	}
-	buf := &bytes.Buffer{}
-	err = template.Execute(buf, config)
+
+	// Render raw values
+	values, err := renderRawValues(chrt, &config)
+	if err != nil {
+		return err
+	}
+
+	// Render templates with raw values
+	renderer := engine.New()
+	rendered, err := renderer.Render(chrt, values)
 	if err != nil {
 		return err
 	}
 
+	injectTmpl, err := renderInjectTemplate(rendered, &config)
+	if err != nil {
+		return err
+	}
+
+	injectOptions := newInjectOptions()
+	injectOptions.proxyConfigOptions = options.proxyConfigOptions
+
+	// Special case for linkerd-proxy running in the Prometheus pod.
+	injectOptions.proxyOutboundCapacity[config.PrometheusImage] = prometheusProxyOutboundCapacity
+
+	return InjectYAML(&injectTmpl, w, ioutil.Discard, injectOptions)
+}
+
+func loadChart(config *installConfig) (*chart.Chart, error) {
+	chrt := &chart.Chart{}
+
+	// Load chart metadata
+	if err := loadMetaData(chrt); err != nil {
+		return &chart.Chart{}, err
+	}
+
+	// Always load the base template
+	if err := loadTemplate(chrt, baseTemplateName, baseTemplatePath); err != nil {
+		return &chart.Chart{}, err
+	}
+
 	if config.EnableTLS {
-		tlsTemplate, err := template.New("linkerd").Parse(install.TLSTemplate)
-		if err != nil {
-			return err
-		}
-		err = tlsTemplate.Execute(buf, config)
-		if err != nil {
-			return err
+		if err := loadTemplate(chrt, tlsTemplateName, tlsTemplatePath); err != nil {
```

Comment:
It'll be important from a chart perspective to handle all the configuration inside of the chart and not load separate templates.

Reply:
This is fixed.